File: src/0415-add-strings/solution.go
package solution

import (
	"fmt"
	"strconv"
	"strings"
)

func s2i(v string) int {
	r, _ := strconv.Atoi(v)
	return r
}

func i2s(v int) string {
	return fmt.Sprintf("%v", v)
}

func addStrings(num1 string, num2 string) string {
	// 补全前置位 0, 使位数对齐
	l1, l2 := len(num1), len(num2)
	if l1 > l2 {
		num2 = strings.Repeat("0", l1-l2) + num2
	} else {
		num1 = strings.Repeat("0", l2-l1) + num1
	}

	ret := ""
	in := 0
	for i := len(num1) - 1; i >= 0; i-- {
		tmp := s2i(string(num1[i])) + s2i(string(num2[i])) + in
		in = tmp / 10
		ret = i2s(tmp%10) + ret
	}
	if in > 0 {
		ret = "1" + ret
	}

	return ret
}

File: src/0415-add-strings/solution.js
/**
 * @param {string} num1
 * @param {string} num2
 * @return {string}
 */
var addStrings = function(num1, num2) {
    var num1_len = num1.length;
    var num2_len = num2.length;
    for (var i = 0; i < Math.abs(num2_len - num1_len); i++) {
        if (num2_len > num1_len) {
            num1 = '0' + num1;
        } else {
            num2 = '0' + num2;
        }
    }
    var carry = 0;
    var res = '';
    for (var i = num1.length - 1; i >= 0; i--) {
        var tmp = parseInt(num1[i]) + parseInt(num2[i]) + carry;
        carry = tmp >= 10 ? 1 : 0;
        res = tmp % 10 + res;
    }
    if (carry) {
        res = '1' + res;
    }
    return res;
}

module.exports = addStrings;
